avocado/core/test.py

```diff
@@ -702,7 +703,9 @@ class SimpleTest(Test):
     def __init__(self, name, params=None, base_logdir=None, job=None):
         super(SimpleTest, self).__init__(name=name, params=params,
                                          base_logdir=base_logdir, job=job)
-        self._command = self.filename
+        self._command = None
+        if self.filename is not None:
+            self._command = pipes.quote(self.filename)
```

Comment:
Yep, this looks quite good, but there is one uncovered aspect and that is external runner with spaces. Currently it supports arguments, but in an odd way so it always fails (when using spaces in binary name). I'd suggest this version:

```diff
diff --git a/avocado/core/loader.py b/avocado/core/loader.py
index c380aa9..5e26237 100644
--- a/avocado/core/loader.py
+++ b/avocado/core/loader.py
@@ -790,7 +790,7 @@ class ExternalLoader(TestLoader):
             if len(external_runner_and_args) > 1:
                 executable = external_runner_and_args[0]
             else:
-                executable = runner
+                executable = external_runner_and_args[0]
             if not os.path.exists(executable):
                 msg = ('Could not find the external runner executable "%s"'
                        % executable)
```

Which unlike normal test name allows arguments. Alternative acceptable solution would be to remove the `shlex.split` and only accept path to the binary (similarly to test-reference). Also please add a selftest for this example.

^^ I tested this by using echo binary copied to `/tmp/a b`, the first variant allows one to run:

```
avocado --show all run --external-runner "/tmp/a\ b aaa" asdf
```

The later would only allow the usage of:

```
avocado --show all run --external-runner "/tmp/a b" asdf
```
while specification of the path would be friendlier. I don't have strong opinion about which version to use. I'm slightly inclined towards the first way (that's why I mentioned it as first) because it allows to group shared arguments to the external runner ("./run -v -s -r"). The second way is on the other hand easier to follow as it's the same behavior as test reference. The reason for me preferring the first approach is that I see external-runner as advanced feature and advanced features should be powerful.